### PR TITLE
Trajectory visualization

### PR DESCRIPTION
**Pull request by RomanDegtyar**
Task #[8438](https://dev.osll.ru/issues/8438) -- draw robot trajectory on map.

### CODE REVIEW

File: src/utils/path_publisher.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+#include <vector>
+
+
+// default parameters (these can be, probably, customized)
+
+#define DEFAULT_PATH_FRAME_ID "/map"
+#define DEFAULT_ROBOT_FRAME_ID "/base_link"
+#define DEFAULT_WORLD_FRAME_ID "/map"
+#define DEFAULT_TOPIC_NAME "/path"
+#define DEFAULT_NODE_NAME "path_publisher"
```
Comment (mapseamoff) [inline src/utils/path_publisher.cpp]: All parameters should be configurable. An example of how to read parameters from a launch file can be found in `src/ros/init_utils.h`.

File: rviz/debug.rviz [hunk 2]
```diff
@@ -18,7 +19,7 @@ Panels:
       - /2D Nav Goal1
       - /Publish Point1
     Name: Tool Properties
-    Splitter Ratio: 0.588679
+    Splitter Ratio: 0.588679016
```
Comment (mapseamoff) [inline rviz/debug.rviz]: Minor changes in `rviz/debug.rviz` should be removed (e.g. replacing 0.588679 by 0.588679016 etc). Only major changes should be included in the commit (like addition of a new display panel in RViz).

File: src/utils/path_publisher.cpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+#define PUB_QUEUE_SIZE 5
+#define TF_WAIT_TIME 3
+
+
+// Returns stamped base_link pose.
+// In case of failure reports an error and does retries.
+// If ros::shutdown is issued, will return a dummy pose object.
+static geometry_msgs::PoseStamped getCurrentPose() {
+	tf::TransformListener tf_listener;
+    tf::StampedTransform stamped_transform;
```
Comment (mapseamoff) [inline src/utils/path_publisher.cpp]: Inconsitent indentations on lines 27/28, 59/60, etc.

File: CMakeLists.txt [hunk 4]
```diff
@@ -136,6 +136,7 @@ add_executable(gmapping src/slams/gmapping/gmapping.cpp)
 add_executable(tiny_slam src/slams/tiny/tiny_slam.cpp)
 add_executable(viny_slam src/slams/viny/viny_slam.cpp)
 add_executable(viny_slam_x src/slams/vinyx/vinyx_slam.cpp)
+add_executable(path_publisher src/utils/path_publisher.cpp)
```
Comment (mapseamoff) [inline CMakeLists.txt]: `path_publisher.cpp` should be moved from `src/utils` to `src/ros`. The `utils` folder contains classes related to the framework itself and `ros` - ROS adapters for the framework.

**Comment by krinkin** [comment]
Please get requested changes done and come back

**Comment by RomanDegtyar** [comment]
The code has been updated. Please check it one more time.

File: src/ros/path_publisher.cpp [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+#include <nav_msgs/Path.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <tf/transform_listener.h>
+#include <vector>
+#include <string>
+#include "init_utils.h"
+
+static double tf_wait_for_transform_duration() {
+  double wait_for_transform_duration;
+  ros::param::param<double>("~ros/tf/wait_for_transform_duration", wait_for_transform_duration, 3.0);
```
Comment (mapseamoff) [inline src/ros/path_publisher.cpp]: I suggest to remove prefixes from parameters names (e.g.  `~ros/tf/wait_for_transform_duration` -> `~wait_for_transform_duration`)

Comment (RomanDegtyar) [inline src/ros/path_publisher.cpp]: OK, will do soon.

Kindly ask you to explain your comment about the prefixes. Are there any related conventions? 

Thanks.

**Comment by RomanDegtyar** [comment]
The code has been updated according to your suggestions.

**Pull request merged**